sdk/src/wallet/derivation_path.rs

```diff
@@ -105,6 +106,29 @@ impl fmt::Display for DerivationPath {
     }
 }
 
+impl FromStr for DerivationPath {
+    type Err = DerivationPathError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
```

Comment:
Why was it introduced? Can `DerivationPath::from_path_str` be used instead, which also takes the scheme into account?

Reply:
For `SchemeType::Ed25519` all path segments must be hardened.

Reply:
`FromStr` was introduced to support the implementation of `Serde::Deserialize` for `DerivationPath`. And this is required in order to be able to store `DerivationPath`s in the wallet's `Store`, which itself is serialized to file storage using Serde.

I see your point. Currently if a user supplies a non-hardened path like `m/44'/877'/0'/0/0`, we implicitly harden it to `m/44'/877'/0'/0'/0'` before commencing processing. Have we considered just rejecting non-hardened Ed25519 paths as is done in one of the implementations linked to from the SLIP10 specification? See https://github.com/elucidsoft/dotnetstandard-bip32/blob/master/dotnetstandard-bip32/BIP32.cs#L65 where they use the regex `^m(\/[0-9]+')+$`.

Reply:
> I guess requiring scheme information to deserialize derivation paths would mean having to tag derivation paths with their scheme types.
I would prefer this solution. I think that explicit storage of the scheme tag might be also useful from the user perspective.